tests/wpt/mozilla/tests/css/stylesheet_media_queries.html

```diff
@@ -14,8 +14,10 @@
   assert_equals(frame.contentWindow.getComputedStyle(element).backgroundColor, "rgb(255, 0, 0)");
   frame.width = "300";
   frameDoc.documentElement.offsetWidth; // Force layout
-  window.requestAnimationFrame(test.step_func_done(function () {
+  window.requestAnimationFrame(() => {
+    window.requestAnimationFrame(test.step_func_done(function () {
```

Comment:
Hm, why do we need 2 rAFs to make this work. That does not seem right.

On related note, why isn't this tests upstreamed to WPT? I do not see anything servo specific.

Reply:
This works without any rAFs in both firefox and chromium:
```
window.onload = test.step_func(function() {
  var frame = document.getElementById("myframe");
  var frameDoc = frame.contentWindow.document;
  var element = frameDoc.getElementById("testelement");
  assert_equals(frame.contentWindow.getComputedStyle(element).backgroundColor, "rgb(255, 0, 0)");
  frame.width = "300";
  frameDoc.documentElement.offsetWidth; // Force layout
  test.step_func_done(function () {
    assert_equals(frame.contentWindow.getComputedStyle(element).backgroundColor, "rgb(0, 255, 0)");
  })()
});
```

Reply:
I suspect there is something wrong about the way we are processing updates here (either the restyle or the media query update) or else Firefox and Chromium have a more optimized implementation. In any case, I think this issue existed before my change.

Reply:
Indeed, we should open issue about that.

Reply:
I've opened #38446 for this.